### ISSUE TITLE
[BUG]: "preinstall": "npx only-allow pnpm" breaks application

### ISSUE THREAD

**Issue by DouglasCalora** [qmhc/vite-plugin-dts#123]
I created a ts library for using in my projects using vite-plugin-dts, but in the last version of `vite-plugin-dts` was added `"strict pnpm": https://github.com/qmhc/vite-plugin-dts/commit/05335e3da6dbb2c1724058acebcb48ffdda143a0`, this breaks my apps that use my libry.

E.g., I have my libry `@bildvitta/store-adapter` that uses vite-plugin-dts, the apps that use `@bildvitta/store-adapter` are breaking because of the changes in `vite-plugin-dts`.

This is the error when I try to run my apps:
![image](https://user-images.githubusercontent.com/32134017/194336911-a903f823-533d-4a57-8378-80eb58646787.png)

**Comment by eliasassuncao** [comment]
up

**Comment by felipegoulart** [comment]
I'm having the same problem!
Why was strict use of pnpm added?

**Comment by qmhc** [comment]
Which `npm` version did you use?

**Comment by DouglasCalora** [comment]
> Which `npm` version did you use?

`6.14.17`

**Comment by felipegoulart** [comment]
`6.14.17`

**Comment by qmhc** [comment]
Well, the intention of this change just limit the package manager in this project.

This problem seems not occur in `npm@8`. But I will also remove the `preinstall` for compatibility with `npm@6`.

**Comment by DouglasCalora** [comment]
Yes in `npm@8` this error not happen, but this is a `BREAKING CHANGE`, thanks for your help!
